cmd/juju/model/defaultscommand.go

```diff
@@ -97,120 +127,425 @@ func (c *defaultsCommand) SetFlags(f *gnuflag.FlagSet) {
+// juju model-defaults us-east-1 --reset bar foo=baz
+// juju model-defaults --reset bar us-east-1 foo=baz
+//
+// If one doesn't specify a cloud or region the command is still valid but for
+// setting the default on the controller. So the following two commands are
+// equivalent.
+// juju model-defaults foo=baz --reset bar
+// juju model-defaults --reset bar foo=baz
+//
+// Of course one can specify multiple keys to reset --reset a,b,c and one can
+// also specify multiple values to set a=b c=d e=f. I.e. comma separated for
+// reset and space separated for setting. One may also only set or reset as a
+// singular action. The last two here are equivalent.
+// juju model-defaults --reset foo
+// juju model-defaults a=b c=d e=f
+// juju model-defaults a=b c=d --reset e,f
+// juju model-defaults --reset e,f a=b c=d
+//
+// cloud/region may also be specified so above examples with that option might
+// be like the following three equivalent invokations.
+// juju model-defaults us-east-1 a=b c=d --reset e,f
+// juju model-defaults us-east-1 --reset e,f a=b c=d
+// juju model-defaults --reset e,f us-east-1 a=b c=d
+//
+// Finally one can also ask for the all the defaults or the defaults for one
+// specific setting. In this case specifying a region is not valid as
+// model-defaults shows the settings for a value at all locations that it has a
+// default set -- or at a minimum the default and  "-" for a controller with no
+// value set.
+// simultaneously.
+// juju model-defaults
+// juju model-defaults no-proxy
+//
+// It is not valid to reset and get or to set and get values. It is also neiter
+// valid to reset and set the same key, nor to set the same key to different
+// values in the same command.
+//
+// For those playing along that all means the first positional arg can be a
+// cloud/region, a region, a key=value to set, a key to get the settings for,
+// or empty. Other caveats are that one cannot set and reset a value for the
+// same key, that is to say keys to be mutated must be unique.
+//
+// Here we go...
 func (c *defaultsCommand) Init(args []string) error {
-	if c.reset {
-		// We're resetting defaults.
-		if len(args) == 0 {
-			return errors.New("no keys specified")
+	var err error
+	//  If there's nothing to reset and no args we're returning everything. So
+	//  we short circuit immediately.
+	if len(args) == 0 && len(c.reset) == 0 {
+		c.action = c.getDefaults
+		return nil
+	}
+
+	// If there is an argument provided to reset, we turn it into a slice of
+	// strings and verify them. If there is one or more valid keys to reset and
+	// no other errors initalizing the command, c.resetDefaults will be called
+	// in c.Run.
+	if err = c.parseResetKeys(); err != nil {
+		return errors.Trace(err)
+	}
+
+	// Look at the first positional arg and test to see if it is a valid
+	// optional specification of cloud/region or region. If it is then
+	// cloudName and regionName or only regionName are set on the object and
```

Comment:
Fixed.  I had originally thought that api filled in the cloud along the way, but learned that was an incorrect assumption and failed to update the documentation.